compiler/rustc/build.rs

```diff
@@ -0,0 +1,28 @@
+use std::env;
+
+fn main() {
+    let target_os = env::var("CARGO_CFG_TARGET_OS");
+    let target_env = env::var("CARGO_CFG_TARGET_ENV");
+    if Ok("windows") == target_os.as_deref() && Ok("msvc") == target_env.as_deref() {
+        set_windows_exe_options();
+    }
+}
+
+// Add a manifest file to rustc.exe and increase the main thread's stack size.
+fn set_windows_exe_options() {
+    const STACK_SIZE: u32 = 2 * 1024 * 1024;
+    static WINDOWS_MANIFEST_FILE: &str = "Windows Manifest.xml";
+
+    let mut manifest = env::current_dir().unwrap();
+    manifest.push(WINDOWS_MANIFEST_FILE);
+
+    println!("cargo:rerun-if-changed={}", WINDOWS_MANIFEST_FILE);
+    // By default reserve `STACK_SIZE` many bytes for new thread, including the main thread).
+    // Note that memory is not actually committed until used.
+    println!("cargo:rustc-link-arg-bin=rustc-main=/STACK:{}", STACK_SIZE);
```

Comment:
The main thread barely needs any memory. Compilation happens on a thread started by rustc_driver.

Reply:
Ah ok. I know other applications have tripped over this but rustc doesn't need it.